src/framework/vst/internal/vstinstancesregister.h

```diff
@@ -72,12 +71,41 @@ class VstInstancesRegister : public IVstInstancesRegister, public async::Asyncab
 private:
     mutable std::mutex m_mutex;
 
-    using FxPluginKey = std::pair<muse::audio::AudioResourceId, muse::audio::AudioFxChainOrder>;
-    using VstFxInstancesMap = std::map<FxPluginKey, IVstInstancePtr>;
+    enum class Type {
+        Undefined = 0,
+        Instrument,
+        Effect
+    };
+
+    struct Key {
+        Type type = Type::Undefined;
+        muse::audio::AudioResourceId resourceId;
+        muse::audio::TrackId trackId;
```

Comment:
These require default values as they're just ints under the hood

Reply:
fixed